src/shared/contexts/query.tsx

```diff
@@ -316,11 +361,105 @@ export const QueryProvider: FC = ({children}) => {
       headers['Authorization'] = `Token ${override.token}`
     }
 
-    const body = {
+    const body: RequestBody = {
       query,
       dialect: {annotations: ['group', 'datatype', 'default']},
     }
 
+    if (mechanism !== OverrideMechanism.Injection) {
+      const options: Record<string, any> = {}
+
+      if (Object.keys(override?.vars ?? {}).length) {
+        options.v = override.vars
+      }
+      if (Object.keys(override?.params ?? {}).length) {
+        options.params = override.params
+      }
+      if (Object.keys(override?.task ?? {}).length) {
+        options.task = override.task
+      }
+
+      const optionTexts = Object.entries(options)
+        .map(([k, v]) => {
+          const vals = Object.entries(v).map(([_k, _v]) => `  ${_k}: ${_v}`)
+          return `option ${k} =  {${vals.join(',\n')}}`
+        })
+        .join('\n\n')
+
+      const queryAST = parse(query)
+      let optionAST = parse(optionTexts)
```

Comment:
An observation. Not something to change now. 

Because we parse the `option v = {...}` into an AST, and then modify/unify it...we are technically including AST metadata (e.g. location, source) in the extern option sent with the network payload:
<img width="404" alt="Screen Shot 2022-08-03 at 12 49 13 PM" src="https://user-images.githubusercontent.com/10232835/182711534-935f75ac-a665-42c4-b9ca-cad04b032366.png">

This is different from the old DataExplorer and Dashboards, which construct all of the these AST nodes without parsing them. And therefore the metadata is not included in the extern.
```
const variables: Variable[] = []  // typed objects which shared QueryContext does not use 
const extern = buildUsedVarsOption(text, variables)
```

Reply:
Also a note to @rockstar . This is an additional point at which the code will need to update for this issue to remove parse() (https://github.com/influxdata/ui/issues/5169). In order to remove the use of parse here, we have to construct our extern object.

In old DataExplorer and Dashboards, they have utils methods which utilize the Variable objects in order to construct an AST. Since shared QueryContext does not utilize the Variable types (for reasons that @drdelambre is most familiar about)...we just have to create our own helper methods here.